pkg/storage/stores/indexshipper/compactor/compactor.go

```diff
@@ -343,6 +370,27 @@ func (c *Compactor) starting(ctx context.Context) (err error) {
 	}
 	level.Info(util_log.Logger).Log("msg", "compactor is ACTIVE in the ring")
 
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO: This goes in your service
+func (c *Compactor) sizeBasedCompaction(ctx context.Context) error {
+	if c.cfg.SharedStoreType != config.StorageTypeFileSystem {
+		return nil
+	}
+
+	// TODO: Find the chunks you want to delete
+	// Use the stuff you wrote before
+	// Make sure you use the absolute path
+	// Turn your paths into chunk refs with index_util.ParseChunkRef(index_util.DecodeKey([]byte("absolute path")))
+	// Pass the list of to delete to RunSizeCompaction
```

Comment:
Hi @MasslessParticle 

Do you mean something like this?

```golang
var toDelete []retention.ChunkRef

...

fullPath := directory + "/" + file.Name()
chunkRef, _, err := compactor_util.ParseChunkRef(compactor_util.DecodeKey([]byte(fullPath)))
level.Info(util_log.Logger).Log("msg", "block size retention exceeded, removing file", "filepath", fullPath)
if err != nil {
    level.Error(util_log.Logger).Log("msg", err.Error())
    continue
}

toDelete = append(toDelete, chunkRef)
```

If so, with this example `ParseChunkRef()` returns an error....

Reply:
What's `fullPath` look like and what's the error?